pkg/prometheus/promcfg.go

```diff
@@ -3409,3 +3410,22 @@ func (cg *ConfigGenerator) getScrapeClassOrDefault(name *string) *monitoringv1.S
 	}
 	return nil
 }
+
+func getLowerByteSize(v *monitoringv1.ByteSize, cpf *monitoringv1.CommonPrometheusFields) *monitoringv1.ByteSize {
```

Comment:
My rationale for passing the whole object is that it reduces the risk of bug since the parameters have different types. Otherwise it's easy to invert the user value and the enforced value.